### PR TITLE
[Modules] Added Key Module to SQL Server

### PR DESCRIPTION
**Pull request by fabmas**
# Description

Updated API versions
Added Key Module to SQL Server

| Pipeline |
| - |
|[Sql: Servers](https://github.com/Azure/ResourceModules/actions/workflows/ms.sql.servers.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/Microsoft.Sql/servers/.test/common/dependencies.bicep [hunk 1]
```diff
@@ -56,11 +59,46 @@ resource privateDNSZone 'Microsoft.Network/privateDnsZones@2020-06-01' = {
...
+  resource key 'keys@2022-07-01' = {
+    name: 'keyEncryptionKey'
+    properties: {
+      kty: 'RSA'
+    }
+  }
+}
+
+resource keyPermissions 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid('msi-${keyVault::key.id}-${location}-${managedIdentity.id}-Key-Reader-RoleAssignment')
```
Comment (eriqua) [inline modules/Microsoft.Sql/servers/.test/common/dependencies.bicep:88]:
```suggestion
  name: guid('msi-${keyVault::key.id}-${location}-${managedIdentity.id}-Key-Vault-Crypto-Service-Encryption-User-RoleAssignment')
```

File: modules/Microsoft.Sql/servers/.test/common/dependencies.bicep [hunk 2]
```diff
@@ -56,11 +59,46 @@ resource privateDNSZone 'Microsoft.Network/privateDnsZones@2020-06-01' = {
...
+  scope: keyVault // keyVault::key
```
Comment (eriqua) [inline modules/Microsoft.Sql/servers/.test/common/dependencies.bicep:89]: Should this be done at key vault or key vault key scope? If the key is intended, then it scould be changed back to 
```suggestion
  scope: keyVault::key
```

Comment (fabmas) [reply modules/Microsoft.Sql/servers/.test/common/dependencies.bicep]: you are right, it should be done at Key scope! (Tested just now)

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**